tensorflow_addons/callbacks/tqdm_progress_bar.py

```diff
@@ -115,9 +121,81 @@ def on_train_begin(self, logs=None):
         self.total_steps = self.params["steps"]
 
     def on_train_end(self, logs={}):
+        self.is_training = False
         if self.show_overall_progress:
             self.overall_progress_tqdm.close()
 
+    def on_test_begin(self, logs={}):
+        if self.is_training is False:
+            self.test_num_epochs = self.params["epochs"]
+            # set counting mode
+            self.mode = "steps"
+            self.test_total_steps = self.params["steps"]
+            if self.show_epoch_progress:
+                self.test_epoch_progress_tqdm = self.tqdm(
+                    total=self.test_total_steps,
+                    desc="Evaluating",
+                    bar_format=self.epoch_bar_format,
+                    leave=self.leave_epoch_progress,
+                    dynamic_ncols=True,
+                    unit=self.mode,
+                )
+            self.test_num_samples_seen = 0
+            self.test_steps_to_update = 0
+            self.test_steps_so_far = 0
+            self.test_logs = defaultdict(float)
+
+    def on_test_batch_end(self, batch, logs={}):
```

Comment:
Follow up from the same comment for reusing variables instead of creating `test_*` variables, if we are able to re-use those variables, we are able to refactor `on_epoch_end|begin` method and extract the inner logic out and call those methods from `on_epoch_*` and `on_test_batch_*` as we can see that there are many duplicating code.